ALICE3/TableProducer/onTheFlyTOFPID.cxx

```diff
@@ -0,0 +1,279 @@
+      0., 0., 0.,
+      0., 0., 0., 0.,
+      0., 0., 0., 0., 0.};
+    float s, c, x;
+    o2::math_utils::sincos(ptetaphi[2], s, c);
+    o2::math_utils::rotateZInv(xyz[0], xyz[1], x, params[0], s, c);
+    params[1] = xyz[2];
+    params[2] = 0.; // since alpha = phi
+    auto theta = 2. * atan(exp(-ptetaphi[1]));
+    params[3] = 1. / tan(theta);
+    params[4] = charge / ptetaphi[0];
+
+    // Initialize TrackParCov in-place
+    new (&o2track)(o2::track::TrackParCov)(x, ptetaphi[2], params, covm);
+  }
+
+  Double_t TrackLength(o2::track::TrackParCov track, Double_t lX0, Double_t lX1, Double_t lMagneticField)
+  {
+    std::array<float, 3> lPointN;
+    std::array<float, 3> lPointNplus;
+    Double_t lLength = 0.0;
+    track.propagateTo(lX0, lMagneticField);
+    for (Int_t iStep = 1; iStep < lNStepsLIntegrator; iStep++) {
+      track.getXYZGlo(lPointN);
+      Float_t lPosition = lX0 + (lX1 - lX0) * ((Float_t)(iStep)) / ((Float_t)(lNStepsLIntegrator - 1));
+      track.propagateTo(lPosition, lMagneticField);
+      track.getXYZGlo(lPointNplus);
+      lLength += std::hypot(lPointNplus[0] - lPointN[0], lPointNplus[1] - lPointN[1], lPointNplus[2] - lPointN[2]);
+    }
+    return lLength;
+  }
+
+  Double_t Velocity(Double_t lMomentum, Double_t lMass)
+  {
+    // Momentum p and mass m -> returns speed in centimeters per picosecond
+    // Useful for TOF calculations
+    Double_t lA = TMath::Power(lMomentum / lMass, 2);
+    return 0.0299792458 * TMath::Sqrt(lA / (1 + lA));
+  }
+
+  void process(soa::Join<aod::Collisions, aod::McCollisionLabels>::iterator const& collision, soa::Join<aod::Tracks, aod::TracksCov, aod::McTrackLabels> const& tracks, aod::McParticles const&)
+  {
+    o2::math_utils::Point3D<float> pvpos{collision.posX(), collision.posY(), collision.posZ()};
+    std::array<float, 6> pvcov;
+    pvcov[0] = collision.covXX();
+    pvcov[1] = collision.covXY();
+    pvcov[2] = collision.covYY();
+    pvcov[3] = collision.covXZ();
+    pvcov[4] = collision.covXY();
+    pvcov[5] = collision.covXZ();
+    o2::dataformats::VertexBase pvvtx(pvpos, pvcov);
+
+    auto mcCollision = collision.mcCollision();
+    o2::math_utils::Point3D<float> mcpvpos{mcCollision.posX(), mcCollision.posY(), mcCollision.posZ()};
+    std::array<float, 6> mcpvcov; // dummy!
+    for (int ii = 0; ii < 6; ii++)
+      mcpvcov[ii] = 1e-6;
+    o2::dataformats::VertexBase mcpvvtx(mcpvpos, mcpvcov);
+
+    for (auto& track : tracks) {
+      // first step: find precise arrival time (if any)
+      // --- convert track into perfect track
+      if (!track.has_mcParticle()) // should always be OK but check please
+        LOG(error) << "Oh no! No mcParticle label for this track! This shouldn't happen!";
+
+      o2::track::TrackParCov o2track;
+      auto mcParticle = track.mcParticle();
+      convertMCParticleToO2Track(mcParticle, o2track);
+
+      float lX_PV = -100, lX_iTOF = -100, lX_oTOF = -100, lThisTrackLength_iTOF = -1, lThisTrackLength_oTOF = -1;
```

Comment:
The variables names should be improved and underscores removed.

Reply:
I would prefer to keep the underscores, and the names are as informative as they should be in the context of the code. For instance, the first variables correspond to Track X at inner and at outer TOF, and the underscore kind of serves exactly to highlight that. I considered doing "XAtiTOF" but it's just confusing without the underscores...

Reply:
Underscores are allowed only in the names of namespaces. Names of variables follow the [camelCase convention](https://rawgit.com/AliceO2Group/CodingGuidelines/master/naming_formatting.html?showone=Variable_Names#Variable_Names).
I think that `xPv`, `xITof` (`xITOF`), `xOTof` (`xOTOF`) are pretty clear.

Reply:
OK! Changed, thanks!